### PR TITLE
add job id cat, ls, and tree commands

### PR DESCRIPTION
**Pull request by j-ogas**
Work in progress.

Addresses #46.

TODO:
- [x] determine what to do (if anything) about sub directories and files. For example, `pav_cfg.working_dir/'tests'/job_id/` has a `build` directory with files. Do we care?
- [x] add try-catch block around reading, i.e., guard against directory `--filename` arguments
- [ ] add travis tests
- [ ] make travis happy

### CODE REVIEW

File: lib/pavilion/plugins/commands/file.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import sys
+
+from pavilion import commands
+from pavilion import utils
+
+class FileCommand(commands.Command):
+
+    def __init__(self):
+        super().__init__(
+            'file',
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/commands/file.py]: Let's call this the 'cat' command, and have an 'ls' and 'file' alias for it.

File: lib/pavilion/plugins/commands/file.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        )
+        parser.add_argument(
+            '--filename',
+            type=str,
+            help="name of a file within working_dir/tests/<job id>/"
+        )
+
+    def run(self, pav_cfg, args):
+
+        test_dir = pav_cfg.working_dir/'tests'
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/commands/file.py]: The name of the test directory changed to `pav_cfg.working_dir/'test_runs'`

File: lib/pavilion/plugins/commands/file.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        else:
+            for file_ in os.listdir(job_dir):
+                utils.fprint(file_, file=sys.stdout)
+
+        return 0
+
+
+def print_file(filename):
+    try:
+        with open(filename, 'r') as file_:
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/commands/file.py]: `file` isn't a _builtin_ anymore. You can use it as a variable name.

File: lib/pavilion/plugins/commands/file.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+            while True:
```
Comment (Paul-Ferrell) [inline lib/pavilion/plugins/commands/file.py]: This doesn't matter here, but starting in python3.8:
```python
while block := file.read(4096):
  utils.fprint(block, file=sys.stdout)
```

**Comment by Paul-Ferrell** [comment]
Check out the the tree command. 
We don't have to do anything that fancy, but just a basic tree would be cool.

**Comment by Paul-Ferrell** [comment]
And it would be really nice to have a way to get metadata, maybe from a `-l` option?

File: lib/pavilion/plugins/commands/file.py (hunk 1 above)
Comment (j-ogas) [reply lib/pavilion/plugins/commands/file.py]: I wonder if this should just be split into two separate plugin commands, e.g., `commands/cat.py` and `commands/ls.py`?

**Comment by j-ogas** [comment]
example of `pav ls`:
```bash
$ ./bin/pav ls 0000229
Could not find a pavilion config file. Using an empty/default config.
config
status
build.sh
build_hash
run.tmpl
build/
  stream.c -> /home/jogas/git/pavilion2/stream.c
  pav_build_log -> /home/jogas/git/pavilion2/pav_build_log
  stream -> /home/jogas/git/pavilion2/stream
  .built_by -> /home/jogas/git/pavilion2/.built_by
jobid
slurm.log
kickoff.log
```

One thing we need to discuss about `pav cat` is how to pass a filename. Using the example above, what would be the correct way to specify a file that belongs to a non-top level directory? For example: `0000229/build/pav_build_log`. My thoughts are:

1) `pav cat 0000229 build/pav_build_log`; or
2) `pav cat 0000229 pav_build_log`

Option one is much easier to implement. Option two is cleaner but introduces a corner case where there may be files with the same name present in different level directories. What are your thoughts @pflarr ?

**Pull request merged**